### PR TITLE
service scheduling functions

### PR DESCRIPTION
**Pull request by joseph-hunter**

### CODE REVIEW

File: service_data_fetch.js [hunk 1]
```diff
@@ -190,7 +190,53 @@ async function getConflicts() {
   return scheduleConflicts;
 }
 
+async function getNeededPositions() {
+  const neededPositions = {};
+  const services = await getServices();
+  const SERVICE_ID = services["Current Plan"].id;
```
Comment (dino-rodriguez) [inline service_data_fetch.js]: A little odd to have this in all caps if its not a global const

File: service_data_fetch.js [hunk 2]
```diff
@@ -190,7 +190,53 @@ async function getConflicts() {
...
+  ) {
+    console.log("All positions are filled!");
+  } else {
+    console.log("Positions still needed: ", neededPositions);
+  }
+  return neededPositions;
+}
+
 const services = {
+  createPreferences: function (people) {
```
Comment (dino-rodriguez) [inline service_data_fetch.js:213]: Why are you assigning this function as a value in this dictionary?

File: service_data_fetch.js [hunk 3]
```diff
@@ -242,3 +288,5 @@ const services = {
 };
 
 module.exports = services;
+
+getNeededPositions();
```
Comment (dino-rodriguez) [inline service_data_fetch.js]: Is this meant to be here?

**Review by dino-rodriguez — APPROVED** [review]
Just a few questions - not totally sure how this fits into the greater code, but it seems reasonable. Going ahead and approving

File: service_data_fetch.js (hunk 2 above)
Comment (joseph-hunter) [reply service_data_fetch.js:213]: Fixing this

**Pull request merged**